dotCMS/src/main/java/com/dotmarketing/portlets/rules/actionlet/PersonaActionlet.java

```diff
@@ -61,27 +62,39 @@ public Instance instanceFrom(Map<String, ParameterModel> parameters) {
     }
 
     @Override
-    public boolean evaluate(HttpServletRequest request, HttpServletResponse response, Instance instance) {
+    public boolean evaluate(final HttpServletRequest request, final HttpServletResponse response, final Instance instance) {
+
         boolean result;
+
         try {
-            Optional<Visitor> opt = visitorAPI.getVisitor(request);
-            if(opt.isPresent()){
-                User user = userAPI.getSystemUser();
-                Persona p = personaAPI.find(instance.personaId, user, false);
-                if(p == null){
+
+            final Optional<Visitor> opt = visitorAPI.getVisitor(request);
+
+            if(opt.isPresent()) {
+
+                final User user         = userAPI.getSystemUser();
+                final PageMode pageMode = PageMode.get(request);
+                final Persona persona   = pageMode.showLive?
+                        personaAPI.findLive(instance.personaId, user, false):
```

Comment:
Should be
`personaAPI.findLive(instance.personaId, user, pageMode.respectAnonPerms):`